datasource/src/main/scala/quasar/plugin/googlebigtable/datasource/CallbackHandler.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.plugin.googlebigtable.datasource
+
+import slamdata.Predef._
+
+import cats.effect.{ConcurrentEffect, Effect, IO}
+import fs2.Stream
+import fs2.concurrent.Queue
+
+object CallbackHandler {
+  def toStream[F[_]: ConcurrentEffect, A](
+      handler: (Either[Throwable, Option[F[A]]] => Unit) => F[Unit],
```

Comment:
Code here is extracted from an old version of async-blobstore. 

Not sure if it's worth to change but I guess the signature of `handler` in this case can be: `(Either[Throwable, Option[A]] => Unit) => F[Unit]` (i.e. `A` instead of `F[A]`).

Also might be handy to move this to quasar, although AFAIK this is the only usage of this pattern right now so we can also wait until there's another usage.

Reply:
A couple of ideas about this
+ Maybe move this to evaluator module (with `Observer`) since they're used only there, and seem to be implementation detail
+ What do you think about adding newtype or type synonim around this continuation?

Reply:
Happy to move `Observer` into `Evaluator`. 

I like to leave `CallbackHandler` separate because this is generic code, whereas the other code is specific.

> What do you think about adding newtype or type synonim around this continuation?

I was hesitating about this as well. WDYT about something like: 
```
type Callback[F[_], A] = Either[Throwable, Option[F[A]]] => Unit
type Handler[F[_], A] = Callback[F, A] => F[Unit]
```

Reply:
Is `Handler` a `Continuation` actually?

Reply:
> Is Handler a Continuation actually?

Not sure :) I like `Handler` better here.